pkg/statistics/table.go

```diff
@@ -122,53 +128,50 @@ func (m *ColAndIdxExistenceMap) HasAnalyzed(id int64, isIndex bool) bool {
 }
 
 // InsertCol inserts a column with its meta into the map.
-func (m *ColAndIdxExistenceMap) InsertCol(id int64, info *model.ColumnInfo, analyzed bool) {
-	m.colInfoMap[id] = info
+func (m *ColAndIdxExistenceMap) InsertCol(id int64, analyzed bool) {
 	m.colAnalyzed[id] = analyzed
 }
 
-// GetCol gets the meta data of the given column.
-func (m *ColAndIdxExistenceMap) GetCol(id int64) *model.ColumnInfo {
-	return m.colInfoMap[id]
-}
-
 // InsertIndex inserts an index with its meta into the map.
-func (m *ColAndIdxExistenceMap) InsertIndex(id int64, info *model.IndexInfo, analyzed bool) {
-	m.idxInfoMap[id] = info
+func (m *ColAndIdxExistenceMap) InsertIndex(id int64, analyzed bool) {
 	m.idxAnalyzed[id] = analyzed
 }
 
-// GetIndex gets the meta data of the given index.
-func (m *ColAndIdxExistenceMap) GetIndex(id int64) *model.IndexInfo {
-	return m.idxInfoMap[id]
-}
-
 // IsEmpty checks whether the map is empty.
 func (m *ColAndIdxExistenceMap) IsEmpty() bool {
-	return len(m.colInfoMap)+len(m.idxInfoMap) == 0
+	return len(m.colAnalyzed)+len(m.idxAnalyzed) == 0
 }
 
 // ColNum returns the number of columns in the map.
 func (m *ColAndIdxExistenceMap) ColNum() int {
-	return len(m.colInfoMap)
+	return len(m.colAnalyzed)
 }
```

Comment:
And also for this three?

Reply:
It might be either false or true. When you initialize, it must be false, but if you just update, then it might be true.